Source/ProgressionSystemRuntime/Public/Data/PSDataAsset.h

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "PSDataAsset.generated.h"
+
+/**
+ * Contains all progression assets used in the module 
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSDataAsset : public UDataAsset
+{
+	GENERATED_BODY()
+
+public:
+	/** Returns the settings data asset. */
+	static const UPSDataAsset& Get();
+
+	/** Returns the Progression Data Table
+	 * @see UProgressionSystemDataAsset::ProgressionDataTableInternal */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	const FORCEINLINE UDataTable* GetProgressionDataTable() const { return ProgressionDataTableInternal; }
+
+	/** Returns a progression menu widget to be displayed in the main menu*/
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE TSubclassOf<class UPSMenuWidget> GetProgressionMenuWidget() const { return ProgressionMenuWidgetInternal; }
+
+	/** Returns a locked progression icon reference */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	FORCEINLINE UTexture2D* GetLockedProgressionIcon() const { return LockedProgressionIconInternal; }
```

Comment:
Always add `class` before pointer types. You missed it for UTexture and UDataTable
```
const FORCEINLINE [class] UDataTable* 
FORCEINLINE [class] UTexture2D* ...
```
Otherwise, it will fail the cook since we try to avoid any unnecessary includes and the same time you did not 'forward' it with 'class'.

Reply:
added